extensions/interactions/LogicProof/static/js/conversion.ts

```diff
@@ -55,3 +55,5 @@ var logicProofConversion = (function() {
     convertToLogicCharacters: convertToLogicCharacters
   };
 }());
+
+export default logicProofConversion;
```

Comment:
Why do we do `export default` here and nowhere else? /cc @kevinlee12 @YashJipkate

Reply:
`export default` is used here and other places where the variable that is exported (in this case `logicProofConversion`) is actually needed.
Just using require wouldn't put the variable in the scope where it is imported.

Reply:
I'm sorry, I don't understand your explanation. When stuff is exported/imported in other files, "export default" is not used currently, but presumably the imports are needed by the file which does the importing, so I'm not sure what you mean by "is actually needed" here.

Let me try asking my question another way perhaps. Can we find a way that doesn't use `export default` here, and that is more in line with the way other files handle imports, so that the codebase is consistent throughout?

Reply:
Actually, I think this approach is better than what we use now, firstly we are using export in files that are migrated to Angular 8, secondly, this cleans up the window object a little bit because now the `logicProofConversion` (and other files) have closed scope that can be exposed by importing the file.

Reply:
@seanlip I think the main cause for using `export default` here is that the variable that is being exported is needed in the `scope` which it is being imported in.
We could have also exported it without the default but since it's only one variable that is getting exported, then it makes sense to use `export default`.

Reply:
OK -- if this is going to be a general approach, I'm fine with that, but we should have clear general rules on when it should be used. Can these be added to our webpack wiki page and coding style guide? Then, if this is in accordance with those rules, I have no further concerns.

Please add a link to those sections of the wiki pages once they've been added.

Reply:
Hi the wiki page is changed https://github.com/oppia/oppia/wiki/Coding-style-guide#exporting-variables-and-functions-from-a-typescript-file-to-be-imported-in-another-typescript-file

Reply:
Thanks @vojtechjelinek. One question:

"If the file adds variable to the global scope we want to isolate the scope, you should export the variable that is added to the global scope."

Could you please clarify this a bit more? I'm wondering if there's a word missing between "adds variable to the global scope" and "we want to isolate the scope" -- the sentence, as is, does not scan.